packages/hint/src/lib/analyzer.ts

```diff
@@ -60,6 +60,7 @@ const validateConnector = (configuration: Configuration) => {
  */
 export class Analyzer {
     private configuration: Configuration;
+    private engine!: Engine;
```

Comment:
```suggestion
    private engine: Engine;
```

You are already checking it exists before using it so this shouldn't be needed I think.

Reply:
`engine` is not initialized in the constructor, that's why we need it.

Reply:
If that's the case then we definitely shouldn't have this and should be verifying `this.engine` actually exists before using it. Use `private engine?: Engine` instead. You can clean up repeated access by temporarily assigning to a local `engine` variable and testing that (which will cause TypeScript to narrow the type).

Reply:
That is what I usually do, but I so that code somewhere in the project and I thought it was the "right" way to do it :s

Reply:
Fixed!

Reply:
hehe, we definitely have a few uses of `!` floating around I'd love to get rid of when we get the chance 😁